solidity/test/router.test.ts

```diff
@@ -46,191 +39,146 @@ describe.skip('Router', async () => {
-      await expect(router.mustHaveRemoteRouter(origin)).to.be.revertedWith(
-        `No router enrolled for domain. Did you specify the right domain ID?`,
+  describe('#domains', () => {
+    it('returns the domains', async () => {
+      await router.enrollRemoteRouters(
+        [origin, destination],
+        [
+          addressToBytes32(nonOwner.address),
+          addressToBytes32(nonOwner.address),
+        ],
       );
-      await router.enrollRemoteRouter(origin, addressToBytes32(remote));
-      expect(await router.isRemoteRouter(origin, remoteBytes)).to.equal(true);
-      expect(await router.mustHaveRemoteRouter(origin)).to.equal(remoteBytes);
+      expect(await router.domains()).to.deep.equal([origin, destination]);
     });
+  });
+
+  it('non-owner cannot enroll remote router', async () => {
+    await expect(
+      router
+        .connect(nonOwner)
+        .enrollRemoteRouter(origin, addressToBytes32(nonOwner.address)),
+    ).to.be.revertedWith(ONLY_OWNER_REVERT_MSG);
+  });
 
-    it('owner can enroll remote router using batch function', async () => {
-      const remote = nonOwner.address;
-      const remoteBytes = addressToBytes32(nonOwner.address);
-      expect(await router.isRemoteRouter(origin, remoteBytes)).to.equal(false);
-      await expect(router.mustHaveRemoteRouter(origin)).to.be.revertedWith(
-        `No router enrolled for domain. Did you specify the right domain ID?`,
+  describe('#dispatch', () => {
+    let payment: BigNumberish;
+
+    beforeEach(async () => {
+      // Enroll a remote router on the destination domain.
+      // The address is arbitrary because no messages will actually be processed.
+      await router.enrollRemoteRouter(
+        destination,
+        addressToBytes32(nonOwner.address),
+      );
+      const recipient = addressToBytes32(router.address);
+      payment = await mailbox['quoteDispatch(uint32,bytes32,bytes)'](
+        destination,
+        recipient,
+        body,
       );
-      await router.enrollRemoteRouters([origin], [addressToBytes32(remote)]);
-      expect(await router.isRemoteRouter(origin, remoteBytes)).to.equal(true);
-      expect(await router.mustHaveRemoteRouter(origin)).to.equal(remoteBytes);
     });
 
-    describe('#domains', () => {
-      it('returns the domains', async () => {
-        await router.enrollRemoteRouters(
-          [origin, destination],
-          [
-            addressToBytes32(nonOwner.address),
-            addressToBytes32(nonOwner.address),
-          ],
-        );
-        expect(await router.domains()).to.deep.equal([origin, destination]);
-      });
+    it('dispatches a message', async () => {
+      await expect(
+        router.dispatch(destination, body, { value: payment }),
+      ).to.emit(mailbox, 'Dispatch');
     });
 
-    it('non-owner cannot enroll remote router', async () => {
+    it('reverts on insufficient payment', async () => {
```

Comment:
I guess this is kinda implied but might be nice to check for event emission, or some kind of other indication that the hook was correctly passed

Reply:
ditto